ffagtf/min.cpp

```diff
@@ -0,0 +1,342 @@
+
+            for (; l<nlab; l++) {
+                averages[l] /= (double) nums[l];
+                values[l]   += averages[l];
+            }
+        flagstop=0;
+
+        for (int i = 0; i < n; i++)
+            if (alive[i]) {
+                l = label[i];
+                if (inactivelabel[l] || (averages[l]<=erreur && averages[l]>=-erreur)) {
+                      if (visited[i+2]==1) graph[0][i+2] = 0; //!!!
+                      if (visited[i+2]==0) graph[i+2][1] = 0; //!!!
+                      alive[i] = 0; // noeud d�connect� � l'avenir
+                      inactivelabel[l]=1;
+                } else {
+                    flagstop=1; // on continue
+                    if (visited[i+2]==1) {
+                        graph[0][i+2] -= averages[l]; //!!!
+                        if(graph[0][i+2] < 0){
+                            double tmp = -graph[0][i+2];
+                            graph[0][i+2] = graph[i+2][1];
+                            graph[i+2][1] = tmp;
+                        }
+                    }
+                    if (visited[i+2]==0) {
+                        graph[i+2][1] += averages[l]; //!!!
+                        if(graph[i+2][1] < 0){
+                            double tmp = -graph[i+2][1];
+                            graph[i+2][1] = graph[0][i+2];
+                            graph[0][i+2] = tmp;
+                        }
+                    } // end else
+                } // end for
+            } //end if
+        } while (flagstop);
+
+   free(nextlabel);
+   free(inactivelabel);
+   free(nums);
+   free(averages);
+   for(int i = 0; i < n; i++) Y[i] = (double) values[label[i]];
+}
+
+void soft_thresh(double *Y, const double thresh, const int n){
+    for(int i = 0; i < n; i++){
+        double tmp = max(Y[i] - thresh, 0.0);
+        Y[i] = tmp + min(Y[i]+thresh, 0.0);
+    }
+}
+
+
+int main(int argc, char** argv){
+    double lambda1 = 6;
+    double lambda2 = 3;
+    int *edges1, *edges2;
+    bool big_graph = false;
+    int m, n;
+    double *Y;
+
+    string e_file_name = "./_data/e_mine";
+    string n_file_name = "./_data/n";
+
+    ifstream n_infile(n_file_name);
+    ifstream e_infile(e_file_name);
+    int tm1, tm2, j = 0;
+    double tm3;
+    e_infile >> tm1 >> tm2;
+    m = tm2;
+    n = tm1;
+    Y = new double[n]; // nodes filled
+    edges1 = new int[m];
+    edges2 = new int[m];
+    while(e_infile >> tm1 >> tm2){
+      edges1[j] = tm1;
+      edges2[j++] = tm2;
+    }
+
+    j = 0;
+    while(n_infile >> tm3){
+      Y[j++] = tm3;
+    }
+
+    cout << "Done! "<< "# of nodes: " << n << "; # of edges: " << m << endl;
+
+    clock_t t1, t2;
+    t1 = clock();
+    graph_tv(Y, n, m, edges1, edges2, lambda1, 0.0);
+    soft_thresh(Y, lambda2, n);
+    t2 = clock();
+
+    cout << "time is " << ((float)t2 - (float)t1) / CLOCKS_PER_SEC << endl;
+    ofstream out( "./output_ffa.txt" );
+    for(int i = 0; i < n; i++) out << Y[i] << endl;
+
+  return 0;
+}
```

Comment:
`min.cpp` never gets compiled or run?  If that's the case, it should be removed.

Reply:
removed